### PR TITLE
[BL][APB-282] Adding dsl driven set of scenarios to test the features

### PR DESCRIPTION
**Pull request by howzat**
set out in the RAML spec

### CODE REVIEW

File: it/uk/gov/hmrc/agentclientauthorisation/scenarios/package.scala [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  def getInvitations(): HalResourceHelper = {
+    val response: HttpResponse = new Resource(getClientInvitationUrl, port).get()(hc)
+    require(response.status == 200, s"Couldn't get invitations, response status [${response.status}]")
+    HalTestHelpers(response.json)
+  }
+}
+
+
+object EmbeddedSection {
+
```
Comment (bravesirrobin81) [inline it/uk/gov/hmrc/agentclientauthorisation/scenarios/package.scala]: I like the concept here, but could we bend the existing play-hal classes to our will instead of creating new ones?

File: it/uk/gov/hmrc/agentclientauthorisation/scenarios/AgencyInvitesClientISpec.scala [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import uk.gov.hmrc.play.test.UnitSpec
+
+class AgencyInvitesClientISpec extends UnitSpec with MongoAppAndStubs with Inspectors with Inside with Eventually {
+
+  private implicit val arn = RandomArn()
+  private implicit val agentCode = AgentCode("LMNOP123456")
+  private val mtdClientId: MtdClientId = FakeMtdClientId.random()
+  private val MtdSaRegime: String = "mtd-sa"
+
+  "Agencies can filter on STATUS" in {
```
Comment (bravesirrobin81) [inline it/uk/gov/hmrc/agentclientauthorisation/scenarios/AgencyInvitesClientISpec.scala]: Is it worth considering the use of FeatureSpec rather than UnitSpec?

File: it/uk/gov/hmrc/agentclientauthorisation/scenarios/package.scala [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+import uk.gov.hmrc.agentclientauthorisation.scenarios.HalTestHelpers.HalResourceHelper
+import uk.gov.hmrc.agentclientauthorisation.support.Resource
+import uk.gov.hmrc.play.auth.microservice.connectors.Regime
+import uk.gov.hmrc.play.controllers.RestFormats
+import uk.gov.hmrc.play.http.logging.SessionId
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpResponse}
+import views.html.helper._
+
+import scala.util.Try
+
```
Comment (bravesirrobin81) [inline it/uk/gov/hmrc/agentclientauthorisation/scenarios/package.scala]: I'm a little uncomfortable about the fact that this file has a bunch of unrelated stuff in it, and it's already quite large. But, I appreciate that this might not be it's final form.

**Comment by bravesirrobin81** [comment]
Is the intention that after implementing these, we'd reduce the coverage in the current *ISpec tests? I'm concerned about the maintainability of testing absolutely everything in the application many ways.

**Comment by howzat** [comment]
@bravesirrobin81 - I agree, it might become burdensome (and redundant) to keep both. I'd prefer to remove the other *ISpec tests once I know the coverage is reflected in the scenarios.
